Octokit/Models/Request/IssueRequest.cs

```diff
@@ -121,23 +121,45 @@ public enum IssueFilter
     }
 
     /// <summary>
-    /// The range of states that an issue can be in.
+    /// Range of states for Issues, Milestones and PullRequest API.
+    /// </summary>
+    public enum ItemStateFilter
+    {
+        /// <summary>
+        /// Items that are open.
+        /// </summary>
+        Open,
+
+        /// <summary>
+        /// Items that are closed.
+        /// </summary>
+        Closed,
+
+        /// <summary>
+        /// All the items.
+        /// </summary>
+        All
+    }
+
+    /// <summary>
+    /// Items that are open OR closed
     /// </summary>
     public enum ItemState
     {
         /// <summary>
-        /// Isuses that are open (default).
+        /// Issues that are open
         /// </summary>
         Open,
 
         /// <summary>
-        /// Isuses that are closed.
+        /// Issues that are closed
         /// </summary>
         Closed,
 
         /// <summary>
-        /// All the issues.
+        /// All the issues. The option is Obsolete
         /// </summary>
+        [Obsolete("The value is Obsolete and will be removed in a future release as it is not a valid option for Search, Create and Update operations")]
```

Comment:
As agreed by @shiftkey we can just DROP the `All` option here, no need to mark it `[Obsolete]` anymore